cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -371,19 +379,117 @@ public LabelId apply(final ResultSet rows) {
 
   @Nonnull
   @Override
-  public ListenableFuture<LabelMeta> getMeta(final LabelId uid,
-                                             final LabelType type) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Optional<LabelMeta>> getMeta(final LabelId id,
+                                                       final LabelType type) {
+
+    final ListenableFuture<Optional<Row>> metas = getOptionalMeta(id, type);
+    return transform(metas, new Function<Optional<Row>, Optional<LabelMeta>>() {
+      @Nullable
+      @Override
+      public Optional<LabelMeta> apply(final Optional<Row> result) {
+        if (!result.isPresent()) {
+          return Optional.absent();
+        }
+
+        final Row row = result.get();
+
+        return Optional.of(
+            LabelMeta.create(id, type, row.getString("name"), row.getString("meta"), 0));
+      }
+    });
+
+  }
+
+  @Nonnull
+  ListenableFuture<Optional<Row>> getOptionalMeta(final LabelId id,
+                                                  final LabelType type) {
+
+    final ListenableFuture<List<Row>> rowFuture = checkForDuplicates(id, type,
+        getMetaStatement.bind(toLong(id), type.toValue()));
+
+    return transform(rowFuture, new FirstOrAbsentFunction<Row>());
+  }
+
+  /**
+   * Use this function to check for duplicates.
+   *
+   * @param id The LabelId we were looking for in the statement. For logging purpose.
+   * @param type The type we were looking for in the statement. For logging purpose.
+   * @param statement The statement we wish to execute.
+   * @return Returns a list of rows.
+   */
+  @Nonnull
+  ListenableFuture<List<Row>> checkForDuplicates(final LabelId id,
+                                                 final LabelType type,
+                                                 final BoundStatement statement) {
+
+    final ResultSetFuture future = session.executeAsync(statement);
+
+    return transform(future, new Function<ResultSet, List<Row>>() {
+      @Nullable
```

Comment:
This method will never return `null` so the `@Nullable` annotation should be removed. Same goes for the same annotation in the other `checkDuplicates` method.